### PR TITLE
Last one :*

### PR DESCRIPTION
**Pull request by RiveraDarwin**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -19,11 +19,42 @@ const people = [
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
+
+describe("destruct", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () =>{
+      expect(destruct(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+
+  })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:28]: Test looks good, I would recommend placing your test variables inside your it method just before the expect statement as this is where your test variables are used.

File: code-challenges.test.js [hunk 2]
```diff
@@ -19,11 +19,42 @@ const people = [
...
 
 // b) Create the function that makes the test pass.
 
+// Create an function destruct that takes an array of objects
+// .map to loop through the values of the array
+// declare a variable capLetter equal to the value.name and use .spilt(" ") to split the array.
+// Now use .toUpperCase to capitilize the 0 index of the the .name value using .charAt(0)
+// Add the now capitilized name to the rest of the string using .substring(1)
+//return capLetter and .join and since we mapped the array we can call value.occupation
+//Original was hardcoded and while it returned the expected result it was not dynamic. This solution was found by Earl and he was kind enough to help me work through it.
```
Comment (elysemontano) [inline code-challenges.test.js:48]: I appreciate you adding this in here!  It is always good to look for ways to refactor and keep our code dynamic, and sometimes we need help from others so great job on that!

File: code-challenges.test.js [hunk 3]
```diff
@@ -19,11 +19,42 @@ const people = [
...
+
+const destruct = (array) => {
+  return array.map (value => {
+    let capLetter = value.name.split(" ").map(letter =>{
+      return letter.charAt(0).toUpperCase()+letter.substring(1)})
+    return `${capLetter.join(" ")} is ${value.occupation}.`})
+}
```
Comment (elysemontano) [inline code-challenges.test.js:55]: Definitely like this approach as it accounts for some edge cases as well due to it's dynamic nature.

File: code-challenges.test.js [hunk 4]
```diff
@@ -34,18 +65,75 @@ const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
...
 // b) Create the function that makes the test pass.
 
+// Green PASS
+// remainderThree
+//     ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+//create a function names remainderThree that takes in a mixed array
+//use .filter to return a subset of the array where the value is a typeof number.
+// Now we have an array of numbers so .map through on value where value % 3
+// I was really happy with this solution.
```
Comment (elysemontano) [inline code-challenges.test.js:88]: I agree, this looks great!

File: code-challenges.test.js [hunk 5]
```diff
@@ -34,18 +65,75 @@ const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
...
+
+const remainderThree = (array) =>{
+  return array.filter(v => typeof v === 'number').map(v => v % 3)
+}
+
+console.log(remainderThree(hodgepodge1))
+console.log(remainderThree(hodgepodge2))
```
Comment (elysemontano) [inline code-challenges.test.js:95]: Don't forget that you don't need to console log to invoke your function since your test is invoking the function.

File: code-challenges.test.js [hunk 6]
```diff
@@ -34,18 +65,75 @@ const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
...
+// return sum
+// Issue encountered with scope when I declared sum inside the for loop.
+
+const retNumCubed = (array) => {
+      let cubedArr = array.map(v => v ** 3)
+      sum = 0
+      for (let i = 0; i < cubedArr.length; i++){
+           sum += cubedArr[i]
+      }return sum
+}
```
Comment (elysemontano) [inline code-challenges.test.js:136]: Logic looks great!  Can you think of a way of using a higher-order function instead of a for loop here?

File: interview-questions.md [hunk 7]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: This invloves associations in the Cohort Model you would have a has_many relationship. A cohort has many students.
```
Comment (elysemontano) [inline interview-questions.md:10]: There are multiple parts to the question, make sure to address all parts of a question when answering!

File: interview-questions.md [hunk 8]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: An association is a connection between two Active Record models. Rails supports 6 types of associations.
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: GET,PATCH,DELETE will require a params. In the form of :id.
```
Comment (elysemontano) [inline interview-questions.md:18]: Those are definitely the http verbs associated with the RESTful route, but what are the action names?  Also, one line answers in an interview will not make you stand out.  Oftentimes it can come off as uninterested or unmotivated.  I would highly recommend working on elaborating further on the concepts.

File: interview-questions.md [hunk 9]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
-
-action: "GET"    location: /students          
-
+   <!-- this will call the students controller and return all students  -->
+action: "GET"    location: /students
+       <!-- This will fire up the POST method in the controller -->
```
Comment (elysemontano) [inline interview-questions.md:35]: Post is an HTTP verb, not the standard name for the controller method.  I would recommend revisiting the [RESTful routes](https://github.com/learn-academy-2022-bravo/Syllabus/blob/main/rails/restful-routes-crud.md) section of the syllabus.

File: interview-questions.md [hunk 10]
```diff
@@ -7,44 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 action: "POST"   location: /students       
-
+   <!--This will get the form to create a new student  -->
 action: "GET"    location: /students/new
-
+   <!-- This will call the student with the ID of 2 and return it -->
 action: "GET"    location: /students/2  
-
+   <!-- This will edit the information for the student with an ID of 2 -->
```
Comment (elysemontano) [inline interview-questions.md:41]: A "GET" request won't edit or change information.  This will be some kind of a view.

File: rails-commenting.rb [hunk 11]
```diff
@@ -6,25 +6,28 @@
 
 # FILE: app/controller/blog_posts_controller.rb
 
-# ---1)
+# ---1) The controller method of index is the "R" in CRUD and the http verb associated with index is "GET".
```
Comment (elysemontano) [inline rails-commenting.rb:9]: Make sure to read the directions.  In this case, problem 1 is asking to only describe the following line, which is where class BlogPostsController inherits from ApplicationController and will be where all of the controller methods will be written.

**Review by elysemontano — APPROVED** [review]
Your code looks great Darwin!  Your interview answers were meeting the bare minimum requirement though.  One sentence answers in an interview can be viewed as uninterested or unmotivated.  There is always room to expand on your answer, providing more insight on the definitions and/or practical applications where you have used these concepts.  I would recommend spending some time reviewing the [RESTful routes](https://github.com/learn-academy-2022-bravo/Syllabus/blob/main/rails/restful-routes-crud.md) in the syllabus.
